src/main/java/mate/academy/spring/service/CustomUserDetailsService.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.spring.service;
+
+import java.util.Optional;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        Optional<mate.academy.spring.model.User> userOptional = userService.findByEmail(username);
+        mate.academy.spring.model.User user = userOptional
+                .orElseThrow(() -> new UsernameNotFoundException("User not found!"));
```

Comment:
```suggestion
                .orElseThrow(() -> new UsernameNotFoundException("Cant find user with email: " + username));
```

lets make exception messages more informative